spirekey/src/app/(wallet)/pact/page.tsx

```diff
@@ -0,0 +1,189 @@
+'use client';
+import { useAccounts } from '@/resolvers/accounts';
+import { genesisPrivateKey, genesisPubKey } from '@/utils/constants';
+import { signWithKeyPair } from '@/utils/signSubmitListen';
+import { createTransactionBuilder, IBuilder } from '@kadena/client';
+import {
+  Button,
+  Select,
+  SelectItem,
+  Stack,
+  TextareaField,
+  TextField,
+} from '@kadena/kode-ui';
+import {
+  CardContentBlock,
+  CardFixedContainer,
+  CardFooterGroup,
+} from '@kadena/kode-ui/patterns';
+import { initSpireKey, sign } from '@kadena/spirekey-sdk';
+import { Account } from '@kadena/spirekey-types';
+import { ChainId } from '@kadena/types';
+import { useEffect, useState } from 'react';
+
+const addDataString = (dataString: string) => (builder: IBuilder<any>) => {
```

Comment:
i would extract these functions to a utils file or something, so we can test these.